From 5a38211d731ed9ee9854875c4871700d77f99b3a Mon Sep 17 00:00:00 2001
From: CI Assistant <yossi.ilkanaev@intel.com>
Date: Mon, 6 Aug 2018 21:01:57 +0200
Subject: [PATCH] Merge pull request #64 in SW_UGW/linux from
 bugfix/UGW_SW-21765-sys-tmu-queue1-and-sys-tmu-lookup to master

* commit '6a7a2b19c492609ebaaca025c1399ab8b06edf51':
  UGW-21765: Added the overflow check for seq_printf for "/tmu/queue" print
---
 drivers/net/ethernet/lantiq/tmu/drv_tmu_ll.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/lantiq/tmu/drv_tmu_ll.c b/drivers/net/ethernet/lantiq/tmu/drv_tmu_ll.c
index 4b53b53c7179..d900e2fff04e 100644
--- a/drivers/net/ethernet/lantiq/tmu/drv_tmu_ll.c
+++ b/drivers/net/ethernet/lantiq/tmu/drv_tmu_ll.c
@@ -5017,10 +5017,10 @@ int tmu_queue_raw_dump(struct seq_file *s, int pos)
 	tmu_proc_eqt_print(s, i, 0);
 	if (g_queue_dump == 1) {
 		SEQ_PRINTF(s, "\n");
-		pos++;
-	} else {
-		pos++;
 	}
+	if (seq_has_overflowed(s))
+		return pos;
+	pos++;
 	if ((pos + proc_queue_start_id >= EGRESS_QUEUE_ID_MAX) ||
 	    (pos + proc_queue_start_id > proc_queue_end_id)) {
 
